steps/container_steps.py

```diff
@@ -92,6 +92,13 @@ def start_container_with_args(context, pname="java"):
     context.containers.append(container)
     wait_for_process(context, pname)
 
+@given(u'container is started with entrypoint {entrypoint}')
+def start_container_with_entrypoint(context, entrypoint):
+    container = Container(context.config.userdata['IMAGE'], name=context.scenario.name)
+    container.startWithEntryPoint(entrypoint)
```

Comment:
Instead of using this method just populate kwargs and handle them properly in the `Container` constructor above. I didn't catch it in the previous contribution, it should be done similarly.

Can you try to refactor it?

Reply:
I don't fully understand what are the 'args' here. 
I was under the impression that the args where the parameters passed to the container command:

docker run namespace/image arg1 arg2

Or are these arguments the container run args:

docker run arg1=... arg2=... namespace/image 

If that's the case, I should have been able to use it to change the  entrypoint. However, when I tried that is was not working (passing entrypoint as arg).

Reply:
The container class is our custom class to store all the interesting data.

I guess I was too excited about kwargs. Ignore it for a bit.

You could for example add a new `entrypoint` parameter to the constructor here: https://github.com/cekit/behave-test-steps/blob/46a0ddfadc391572be43c672839ed011e458811d/steps/container.py#L51 and set it in the class.

Then you instantiate it:

` container = Container(context.config.userdata['IMAGE'], name=context.scenario.name, entrypoint=BLAH)`

and use the regular ` container.start()` method without the need for `startWithEntryPoint`.

Is it now more clear?

Reply:
Oh nice! Yes! 
I will refactor the PR today!

Reply:
Done! 

I didn't rebase on purpose, I will when I got it right.